src/db/models/user.model.js

```diff
@@ -172,5 +258,42 @@ module.exports = (sequelize, DataTypes) => {
             });
     };
 
+    // This creates a contact method verification token in the DB based on email and expTime.
+    // eslint-disable-next-line max-len
+    User.prototype.generateVerifyAccountToken = function generateVerifyAccountToken(email, expTime) {
+        return randomBytes(20)
+            .then((buf) => {
+                const token = buf.toString('hex');
+                return token;
+            }).then((token) => {
+                this.contactMethodVerificationToken = token;
+                this.contactMethodToVerify = email;
+                const m = moment.utc();
+                m.add(expTime, 'seconds');
+                this.contactMethodVerificationTokenExpires = m.format();
+                return this.save().then(() => token);
+            });
+    };
+
+    User.prototype.addVerifiedContact = function addVerifiedContact(contactMethodToVerify) {
+        if (this.verifiedContactMethods.includes(contactMethodToVerify)) {
+            // Handle instances where the user contact already exsits in the list of verified
+            // contact methods
+            this.contactMethodVerificationToken = null;
+            this.contactMethodVerificationTokenExpires = null;
+            this.contactMethodToVerify = null;
+            return this.save();
+        }
+        const authorizedUsers = this.verifiedContactMethods;
+        const authorizedUsersLength = authorizedUsers.push(contactMethodToVerify.toString());
+        if (authorizedUsersLength > 0) {
+            this.verifiedContactMethods = authorizedUsers;
+            this.contactMethodVerificationToken = null;
+            this.contactMethodVerificationTokenExpires = null;
+            this.contactMethodToVerify = null;
+        }
```

Comment:
This section
```js
        const authorizedUsers = this.verifiedContactMethods;
        const authorizedUsersLength = authorizedUsers.push(contactMethodToVerify.toString());
        if (authorizedUsersLength > 0) {
            this.verifiedContactMethods = authorizedUsers;
            this.contactMethodVerificationToken = null;
            this.contactMethodVerificationTokenExpires = null;
            this.contactMethodToVerify = null;
        }
```
can probably be replaced with
```js
        this.verifiedContactMethods = this.verifiedContactMethods.concat([contactMethodToVerify.toString()]);
        this.contactMethodVerificationToken = null;
        this.contactMethodVerificationTokenExpires = null;
        this.contactMethodToVerify = null;
```

Reply:
Not equivalent

Would be if
```
this.verifiedContactMethods = authorizedUsersLength # Not `authorizedUsers`
```